vagrant/src/main/java/org/jclouds/vagrant/compute/VagrantComputeServiceAdapter.java

```diff
@@ -0,0 +1,362 @@
+    }
+
+   private void init(File path, String name, Template template) {
+      try {
+         writeVagrantfile(path);
+         initMachineConfig(path, name, template);
+      } catch (IOException e) {
+         throw new RuntimeException(e);
+      }
+   }
+
+   private void writeVagrantfile(File path) throws IOException {
+      path.mkdirs();
+      VagrantUtils.write(new File(path, "Vagrantfile"), getClass().getResourceAsStream("/Vagrantfile"));
+   }
+
+   private void initMachineConfig(File path, String name, Template template) throws IOException {
+      File machines = new File(path, "machines");
+      machines.mkdirs();
+      
+      File machineConfig = new File(machines, name + ".yaml");
+      
+      Charset encoding = Charset.forName("UTF-8");
+      
+      //write the config manually, no need to pull dependencies for now
+      BufferedWriter out;
+      FileOutputStream fileOut = new FileOutputStream(machineConfig);
+      out = new BufferedWriter(new OutputStreamWriter(fileOut, encoding));
+      
+      try {
+         out.write("box: " + template.getImage().getId() + "\n");
+         out.write("memory: " + template.getHardware().getRam() + "\n");
+         out.write("cpus: " + countProcessors(template));
+      } finally {
+         out.close();
+      }
+   }
+
+   private int countProcessors(Template template) {
+      int cnt = 0;
+      for (Processor p : template.getHardware().getProcessors()) {
+          cnt += p.getCores();
+      }
+      return cnt;
+   }
+
+   @Override
+   public Iterable<Hardware> listHardwareProfiles() {
+      Set<Hardware> hardware = Sets.newLinkedHashSet();
+      hardware.add(hardware("micro", 512, 1));
+      hardware.add(hardware("small", 1024, 1));
+      hardware.add(hardware("medium", 2048, 2));
+      hardware.add(hardware("large", 4096, 2));
+      return hardware;
+   }
+
+   private Hardware hardware(String name, int ram, int cores) {
+      return new HardwareBuilder()
+              .ids(name)
+              .hypervisor("VirtualBox")
+              .name(name)
+              .processor(new Processor(cores, 1))
+              .ram(ram)
+              .volume(new VolumeBuilder().bootDevice(true).durable(true).type(Type.LOCAL).build())
+              .build();
+   }
+
+   @Override
+   public Iterable<Box> listImages() {
+      //TODO Include online images
+      return Vagrant.forPath(new File(".")).box().list();
+   }
+
+   @Override
+   public Box getImage(final String id) {
+      return Iterables.find(listImages(),
+            new Predicate<Box>() {
+               @Override
+               public boolean apply(Box input) {
+                  return id.equals(input.getName());
+               }
+            }, null);
+   }
+
+   @Override
+   public Iterable<Location> listLocations() {
+      return ImmutableList.of(
+              new LocationBuilder().id("localhost").description("localhost").scope(LocationScope.HOST).build());
```

Comment:
This should have as parent the location provided by the `JustProvider` supplier.